setup.py

```diff
@@ -9,20 +11,28 @@
 dist = Distribution()
 dist.parse_config_files()
 
+NAME = dist.get_name()  # type: ignore
 
-NAME: str = dist.get_name()  # type: ignore
+# Check if building for Pyodide
+is_pyodide = os.getenv("PYODIDE", "0") == "1"
 
+if is_pyodide:
+    # For pyodide we build a universal wheel that must be pure-python
+    # so we must omit the cython-version of scan.
+    ext_modules = []
+else:
+    ext_modules = [
+        Extension(
+            name="pytensor.scan.scan_perform",
+            sources=["pytensor/scan/scan_perform.pyx"],
+            include_dirs=[numpy.get_include()],
+        ),
+    ]
```

Comment:
Depending on an environment variable to conditionally include an extension is very unorthodox and I don't think that it will be robust in the future. As far as I know, there is no canonical way to conditionally include an extension or not when using pyproject.toml or the older setup.py. However, there is a way to get platform conditional requirements following [this pep](https://peps.python.org/pep-0508/#environment-markers). Maybe we could look at the `sys` information to identify that python is running from the browser instead of having `is_pyodide` conditioned on the environment?

Reply:
I'm extremely pro-declarative config, so I'd really like to do this better.

Unfortunately after taking a brief look through the environment markers, I don't see any particular variables that can help in our situation. This feels not so surprising to me since "universal" vs "platform-specific" seems more like a "build" parameter than an "environment" parameter, making it something to be handled at the build-backend (`setuptools`) level rather than the project-config level. On the same note, the `pyproject.toml` standard doesn't seem to support `ext_modules`. So I don't have any ideas for how we could do this better.

Also, unfortunately using `ext_modules` seems to marry us to `setuptools` which I loathe, but that's out-of-scope for this PR.